docs/src/tutorial.rst

```diff
@@ -0,0 +1,505 @@
+                statusCode: 200,
+                headers: {},
+                body: JSON.stringify(body)
+              };
+              callback(null, response);
+            })
+            .catch(rejectedPromise);
+        }
+      }
+
+      function rejectedPromise(error) {
+        var body = error.stack || JSON.stringify(error, null, 2);
+        var response = makeError(body);
+        callback(null, response);
+      }
+    }
+
+Save it and make sure it builds and creates an empty stack.
+Note that since we haven't wired the function to our app,
+the Lambda file does not appear in the output.
+
+.. code-block:: sh
+
+    npm run build
+    cdk synth
+
+.. _create_apples_service:
+
+Step 3: Create Apples Service
+=============================
+
+Add the |APIGATEWAY|, |LAM|, and |S3| packages to our app.
+
+.. code-block:: sh
+
+    npm install @aws-cdk/aws-apigateway
+    npm install @aws-cdk/aws-lambda
+    npm install @aws-cdk/aws-s3
+
+Create the directory *lib* at the same level as the *bin* and *resources*
+directories.
+
+.. code-block:: sh
+
+    mkdir lib
+
+Create the following Typescript file, *apple_service.ts*,
+in the *lib* directory.
+
+.. code-block:: ts
+
+    import cdk = require('@aws-cdk/cdk');
+    import apigateway = require('@aws-cdk/aws-apigateway');
+    import lambda = require('@aws-cdk/aws-lambda');
+    import s3 = require('@aws-cdk/aws-s3');
+
+    export class AppleService extends cdk.Construct {
+      constructor(parent: cdk.Construct, name: string) {
+        super(parent, name);
+
+        const bucket = new s3.Bucket(this, 'AppleStore');
+
+        const handler = new lambda.Function(this, 'AppleHandler', {
+          runtime: lambda.Runtime.NodeJS610,
+          code: lambda.Code.directory('resources'),
+          handler: 'apples.main',
+          environment: {
+            BUCKET: bucket.bucketName
+          }
+        });
+
+        bucket.grantReadWrite(handler.role);
+
+        const api = new apigateway.RestApi(this, 'apples-api', {
+          restApiName: 'Apple Service',
+          description: 'This service serves apples.'
+        });
+
+        const getApplesIntegration = new apigateway.LambdaIntegration(handler, {
+	  requestTemplates:  { "application/json": '{ "statusCode": "200" }' }
+        });
+
+        api.root.addMethod('GET', getApplesIntegration);   // GET /
+      }
+    }
+
+Save it and make sure it builds and creates a (still empty) stack.
+
+.. code-block:: sh
+
+    npm run build
+    cdk synth
+
+.. _add_service:
+
+Step 4: Add the Service to the App
+==================================
+
+To add the service to our app,
+add the following line of code after the existing **import** statement in
+*my_apple_service.ts*.
+
+.. code-block:: ts
+
+    import apple_service = require('../lib/apple_service');
+
+Add the following line of code at the end of the constructor in *my_apple_service.ts*.
+
+.. code-block:: ts
+
+    new apple_service.AppleService(this, 'Apples');
```

Comment:
Provide more context here as to where this needs to be added? I believe it needs to be added to the stack's constructor right?

Reply:
The text before:

Add the following line of code at the end of the constructor in *my_widget_service.ts*.